jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MimeTypes.java

```diff
@@ -37,6 +37,43 @@
 public class MimeTypes
 {
     static final  Logger LOG = LoggerFactory.getLogger(MimeTypes.class);
+    private static final Index<Locale> KNOWN_LOCALES;
+
+    static
+    {
+        Index.Builder<Locale> builder = new Index.Builder<>();
+        builder = builder.caseSensitive(false);
+        for (Locale locale : Locale.getAvailableLocales())
+            builder.with(locale);
+        KNOWN_LOCALES = builder.build();
+    }
+
+    /**
+     * @param languageDashCountry The language and optional country of the local. For example "en" or "un-uk".
+     * @return The {@link Locale} matching the string or null if a matching Locale is not known;
+     */
+    public static Locale getLocale(String languageDashCountry)
+    {
+        // First try for a direct hit in the known Locale cache
+        Locale locale = KNOWN_LOCALES.get(languageDashCountry);
+        if (locale != null)
+            return locale;
+
+        // failing that, create a new Locale instance to normalize the name
+        String language = HttpField.stripParameters(languageDashCountry);
+        int dash = languageDashCountry.indexOf('-');
+        if (dash < 0)
+            locale = new Locale(language);
+        else
+        {
+            language = languageDashCountry.substring(0, dash).trim();
+            String country = languageDashCountry.substring(dash + 1).trim();
+            locale = new Locale(language, country);
+        }
+
+        // look up normalized name in the known Locale cache.
+        return KNOWN_LOCALES.get(locale.toString());
```

Comment:
I do not understand why a `Locale` is created, then turned into a string, then use that string to look up a locale from an index.

Either there is no need to create the new Locale object, or I'm missing something subtle that's lacking a test and/or some comment.

Reply:
@lorban Read the comments!

Reply:
It is to normalize the name if we miss the first cache try.

Reply:
I think we could do better with java 19

Reply:
Normalize how?

Reply:
uk to GB and other transformations that are magically done inside the Locale class.   I've done code coverage and the test covers all paths.

Reply:
@lorban see the private `convertOldISOCodes` method within the Locale constructor.  That's what I'm accessing.

Reply:
Got it, thanks.